java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S4792_java.html

```diff
@@ -0,0 +1,142 @@
+  hostname. </li>
+  <li> Choose log format which is easy to parse and process automatically. It is important to process logs rapidly in case of an attack so that the
+  impact is known and limited. </li>
+  <li> Check that the permissions of the log files are correct. If you index the logs in some other service, make sure that the transfer and the
+  service are secure too. </li>
+  <li> Add limits to the size of the logs and make sure that no user can fill the disk with logs. This can happen even when the user does not control
+  the logged information. An attacker could just repeat a logged action many times. </li>
+</ul>
+<p>Remember that configuring loggers properly doesn't make them bullet-proof. Here is a list of recommendations explaining on how to use your
+logs:</p>
+<ul>
+  <li> Don't log any sensitive information. This obviously includes passwords and credit card numbers but also any personal information such as user
+  names, locations, etc... Usually any information which is protected by law is good candidate for removal. </li>
+  <li> Sanitize all user inputs before writing them in the logs. This includes checking its size, content, encoding, syntax, etc... As for any user
+  input, validate using whitelists whenever possible. Enabling users to write what they want in your logs can have many impacts. It could for example
+  use all your storage space or compromise your log indexing service. </li>
+  <li> Log enough information to monitor suspicious activities and evaluate the impact an attacker might have on your systems. Register events such as
+  failed logins, successful logins, server side input validation failures, access denials and any important transaction. </li>
+  <li> Monitor the logs for any suspicious activity. </li>
+</ul>
+<p>See</p>
+<ul>
+  <li> <a href="https://cwe.mitre.org/data/definitions/532.html">MITRE, CWE-532</a> - Information Exposure Through Log Files </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/117.html">MITRE, CWE-117</a> - Improper Output Neutralization for Logs </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/778.html">MITRE, CWE-778</a> - Insufficient Logging </li>
+  <li> [OWASP Top 10 2017 Category A3 - Sensitive Data Exposure </li>
+  <li> [OWASP Top 10 2017 Category A10 - Insufficient Logging &amp; Monitoring </li>
+  <li> <a href="https://www.sans.org/top25-software-errors/#cat3">SANS Top 25</a> - Porous Defenses </li>
+</ul>
+<h2>Questionable Code Example</h2>
+<p>This rule supports the following libraries: Log4J, <code>java.util.logging</code> and Logback</p>
+<pre>
+// === Log4J 2 ===
+import org.apache.logging.log4j.core.config.builder.api.ConfigurationBuilderFactory;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.*;
+import org.apache.logging.log4j.core.config.*;
+
+abstract class CustomConfigFactory extends ConfigurationFactory {
+    // ...
```

Comment:
RSPEC should describe this part of the code as `Questionable`, and why it is so

Reply:
Fixed, I removed it by mistake while making the example executable.